py3status/modules/khal_calendar.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+"""
+Displays upcoming khal events.
+
+Configuration parameters:
+    cache_timeout: refresh interval for this module (default 60)
+    config_path: Path to khal configuration file. The default None resolves to /home/$USER/.config/khal/config (default None)
+    date_end: Until which datetime the module searches for events (default 'eod')
+    format: display format for this module (default '{appointments}')
+    output_format: khal conform format for displaying event output (default '{start-time} {title}')
+
+Format placeholders:
+    {appointments} list of events in time range
+
+Requires:
+    khal: https://github.com/pimutils/khal
+
+@author @xenrox
+@license BSD
+
+SAMPLE OUTPUT
+{'full_text': '13:00 Eat lunch'}
+
+"""
+from datetime import datetime
+from re import compile as re_compile
+from khal.settings import get_config
+from khal.cli import build_collection
+from khal.controllers import khal_list
+
+
+class Py3status:
+    """
+    """
+
+    # available configuration parameters
+    cache_timeout = 60
+    config_path = None
+    date_end = "eod"
```

Comment:
What can users do/change with `date_end` config? Seems like this could be hidden/hardcoded?

Reply:
The module shows all events from the current time until `date_end`. By default i put "end of day" or `eod` because that seemed to be the most sane default to me. But maybe some people want to see all events for all workdays or only for the next 4 hours (`+4h`).